ServerCore/Areas/Identity/UserAuthorizationPolicy/IsEventAdmin.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using ServerCore.DataModel;
+
+namespace ServerCore.Areas.Identity.UserAuthorizationPolicy
+{
+    public class IsAdminInEventRequirement : IAuthorizationRequirement
+    {
+        public IsAdminInEventRequirement()
+        {
+        }
+    }
+
+    public class IsAdminInEventHandler : AuthorizationHandler<IsAdminInEventRequirement>
+    {
+        private readonly PuzzleServerContext puzzleContext;
+        private readonly UserManager<IdentityUser> userManager;
+
+        public IsAdminInEventHandler(PuzzleServerContext pContext, UserManager<IdentityUser> manager)
+        {
+            puzzleContext = pContext;
+            userManager = manager;
+        }
+
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+                                                       IsAdminInEventRequirement requirement)
+        {
+            PuzzleUser puzzleUser = PuzzleUser.GetPuzzleUserForCurrentUser(puzzleContext, context.User, userManager);
```

Comment:
This and similar methods are making several database calls synchronously even though they're async methods. Should we have the callees be async? (It's ok to file an issue if necessary)